### PR TITLE
Update in the Benney-Luke demo

### PR DESCRIPTION
**Pull request by kalogirou**
I have added a small paragraph in the end of the BL demo to say that we have used the same weak formulations to study a different problem. I have also updated the references by adding our new paper and modifying some of our previous ones.

### CODE REVIEW

File: demos/demo_references.bib [hunk 1]
```diff
@@ -25,6 +25,14 @@ @InBook{2015:lmscup
                   {http://www1.maths.leeds.ac.uk/~matak/lms-cup2015.pdf}
 }
 
+@Manual{Gidel:2017,
```
Comment (dham) [inline demos/demo_references.bib]: Shouldn't this be an article?

File: docs/source/_static/firedrake-apps.bib [hunk 2]
```diff
@@ -1,3 +1,11 @@
+@Manual{Gidel2017,
```
Comment (dham) [inline docs/source/_static/firedrake-apps.bib]: As above.

File: demos/demo_references.bib (hunk 1 above)
Comment (kalogirou) [inline demos/demo_references.bib]: Yes, but we still don't have volume and page information which are essential in the bibtex entry for articles.

Comment (dorugeber) [inline demos/demo_references.bib]: Pretty sure that's not true.  Various bibtex guides say [for article] "Required fields: author, title, journal, year. Optional fields: volume, number, pages, month, note."

Comment (kalogirou) [inline demos/demo_references.bib]: Ok then I apologise. The ref. Gidel:2017 is an article which is accepted by Nonlinear Processes Geophysics.

File: docs/source/_static/firedrake-apps.bib (hunk 2 above)
Comment (kalogirou) [inline docs/source/_static/firedrake-apps.bib]: Yes, this should be changed to article.

File: demos/demo_references.bib (hunk 1 above)
Comment (kalogirou) [inline demos/demo_references.bib]: Update: The paper has been published in Volume 24, pages 43-60 (doi:10.5194/npg-24-43-2017). Can you make the change?

File: docs/source/_static/firedrake-apps.bib (hunk 2 above)
Comment (kalogirou) [inline docs/source/_static/firedrake-apps.bib]: Please include the final publication info, as above. Thanks.

File: demos/demo_references.bib (hunk 1 above)
Comment (dham) [inline demos/demo_references.bib]: Hi Anna, this pull request is from kalogirou:master. Only you have the access to make this change!

**Comment by wence-** [comment]
Thanks!

**Pull request merged**